st2client/st2client/commands/action.py

```diff
@@ -1348,5 +1350,71 @@ def run_and_print(self, args, **kwargs):
                 execution = format_execution_status(execution)
         except resource.ResourceNotFoundError:
             self.print_not_found(args.id)
-            raise OperationFailureException('Execution %s not found.' % (args.id))
+            raise ResourceNotFoundError('Execution %s not found.' % (args.id))
         return self._print_execution_details(execution=execution, args=args, **kwargs)
+
+
+class ActionExecutionTailCommand(resource.ResourceCommand):
+    def __init__(self, resource, *args, **kwargs):
+        super(ActionExecutionTailCommand, self).__init__(
+            resource, kwargs.pop('name', 'tail'),
+            'A command to tail output of a particular execution.',
+            *args, **kwargs)
+
+        self.parser.add_argument('id', nargs='?',
+                                 metavar='id',
+                                 default='last',
+                                 help='ID of action execution to tail.')
+        self.parser.add_argument('--type', dest='output_type', action='store',
+                                 help=('Type of output to tail for. If not provided, '
+                                      'defaults to all.'))
+        self.parser.add_argument('-h', '--help',
+                                 action='store_true', dest='help',
+                                 help='Print usage for the given command.')
+
+    def run(self, args, **kwargs):
+        pass
+
+    @add_auth_token_to_kwargs_from_cli
+    def run_and_print(self, args, **kwargs):
+        execution_id = args.id
+        output_type = getattr(args, 'output_type', None)
+
+        # Special case for id "last"
+        if execution_id == 'last':
+            executions = self.manager.query(limit=1)
+            execution = executions[0] if executions else None
+            execution_id = execution.id
+        else:
+            execution = self.manager.get_by_id(execution_id, **kwargs)
+
+        if not execution:
+            raise ResourceNotFoundError('Execution  with id %s not found.' % (args.id))
+
+        # Execution has already finished
+        if execution.status in LIVEACTION_COMPLETED_STATES:
+            print('Execution %s has completed.' % (execution_id))
+            return
```

Comment:
So if execution is completed, this tail command will print this message and exit?  I'm kinda expecting this command to work like the linux `tail` command.  So why can't we just display the data in the ActionExecutionOutput and if it's already been garbage collected, then from the final result?

Reply:
It's worth noting that this tail is a bit different because there is no concept of "file has completed" (no one will ever write to it again or similar) on file system.

Having said that, I do think that makes some sense and I was thinking about it before so I will make that change.

In case execution has completed, I will hit the `/output` API endpoint to retrieve whole result (for real-time stuff, it uses event stream API).

Reply:
7e579bb22dc0ff957b7ce24727beceac4fbb4b21